tests/fuzz/Makefile

```diff
@@ -0,0 +1,72 @@
+
+LOCAL_CFLAGS = -I../../include -D_FILE_OFFSET_BITS=64
+LOCAL_LDFLAGS = -L../../library			\
+		-lmbedtls$(SHARED_SUFFIX)	\
+		-lmbedx509$(SHARED_SUFFIX)	\
+		-lmbedcrypto$(SHARED_SUFFIX)
+
+LOCAL_LDFLAGS += -L../../crypto/library
+LOCAL_CFLAGS += -I../../crypto/include
+CRYPTO := ../../crypto/library/
+
+ifndef SHARED
+DEP=$(CRYPTO)libmbedcrypto.a ../../library/libmbedx509.a ../../library/libmbedtls.a
+else
+DEP=$(CRYPTO)libmbedcrypto.$(DLEXT) ../../library/libmbedx509.$(DLEXT) ../../library/libmbedtls.$(DLEXT)
+endif
+
+
+DLEXT ?= so
+EXEXT=
+SHARED_SUFFIX=
+# python2 for POSIX since FreeBSD has only python2 as default.
+PYTHON ?= python2
+
+# Zlib shared library extensions:
+ifdef ZLIB
+LOCAL_LDFLAGS += -lz
+endif
+
+ifdef FUZZINGENGINE
+LOCAL_LDFLAGS += -lFuzzingEngine
+endif
+
+# A test application is built for each suites/test_suite_*.data file.
+# Application name is same as .data file's base name and can be
+# constructed by stripping path 'suites/' and extension .data.
+APPS = $(basename $(wildcard fuzz_*.c))
+
+# Construct executable name by adding OS specific suffix $(EXEXT).
+BINARIES := $(addsuffix $(EXEXT),$(APPS))
+
+.SILENT:
+
+.PHONY: all check test clean
+
+all: $(BINARIES)
+
+$(DEP):
+	$(MAKE) -C ../../library
+
+C_FILES := $(addsuffix .c,$(APPS))
+
+%.o: %.c
+	$(CC) $(LOCAL_CFLAGS) $(CFLAGS) -c $<	-o $@
+
+
+ifdef FUZZINGENGINE
+$(BINARIES): %$(EXEXT): %.o common.o $(DEP)
+	echo " $(CC) common.o $< $(LOCAL_LDFLAGS) $(LDFLAGS) -o $@"
+	$(CXX) common.o $<	$(LOCAL_LDFLAGS) $(LDFLAGS) -o $@
```

Comment:
It seems that we have the expectation that `make CFLAGS=-m32` is a correct way of producing 32-bit libraries and executables on a 64-bit machine. This doesn't work here because `-m32` would also need to be in `LDFLAGS`.

I don't think that our expectation is correct: to me it makes sense that you need to write `make CFLAGS=-m32 LDFLAGS=-m32`. We're currently getting away with `make CFLAGS=-m32` because all the places that build an executable build it in a single step from `.c` to the executable. But of course it's perfectly fine to have an intermediate step `.c` → `.o`, and then the `.o` → executable step doesn't use `CFLAGS`. So I can't fault the makefile here.

This means changing the three places in `tests/scripts/all.sh` that build with `CFLAGS` containing `-m32` or `-mx32` to also pass an `LDFLAGS` with the `-m` option. This is my preferred resolution. However, I need to check further to see if there are other places or people who expect that `make CFLAGS=-m32` just works, in which case the linker command line would also need to use `$(CFLAGS)`.

Reply:
I agree with your analysis : you need to write `make CFLAGS=-m32 LDFLAGS=-m32`

Let me know

Reply:
Done in https://github.com/ARMmbed/mbedtls/pull/2684